build.sbt

```diff
@@ -1,3 +1,4 @@
+import sbt.internal.inc.ScalaInstance
```

Comment:
The `internal` makes me nervous.  Is this the canonical way in the latest SBT?

Reply:
lol, sbt-spiewak [does](https://github.com/djspiewak/sbt-spiewak/blob/69f1ee5e1252ba8fc0f447589bd73b8155addef6/core/src/main/scala/sbtspiewak/SpiewakPlugin.scala#L123):

```scala
Def.derive(isDotty := scalaVersion.value.startsWith("3.")))
```

Reply:
I agree, probably using SBT's internals not a good idea. I've asked about the canonical `isDotty` on Gitter :)

Reply:
you can `CrossVersion.partialVersion(scalaVersion.value) match { case Some((2, _)) => ... }`

Reply:
That's probably more robust than `.startsWith("3.")`, but kind of a drag to copy into every little microlibrary.

I'm coming around to the internal status quo: if it ever breaks, it'll break at SBT load time.

Reply:
> lol, sbt-spiewak [does](https://github.com/djspiewak/sbt-spiewak/blob/69f1ee5e1252ba8fc0f447589bd73b8155addef6/core/src/main/scala/sbtspiewak/SpiewakPlugin.scala#L123):

I'm sure you'll be happy to learn that I wrote that :wink:

Reply:
> I agree, probably using SBT's internals not a good idea. I've asked about the canonical `isDotty` on Gitter :)

Unfortunately, I haven't got an answer to my question about canonical `isDotty` on SBT's Gitter :(